path_roles_test.go

```diff
@@ -44,6 +47,10 @@ func TestTokenRole(t *testing.T) {
 		require.NoError(t, err)
 
 		resp, err = testTokenRoleRead(t, b, s)
+		if resp == nil {
+			t.Fatalf("Error: received nil response")
+		}
```

Comment:
This and the below nil check were also causing panics when some environment variables were not set. Would it be appropriate to add to this fatal message "Check if all required environment variables are set"?

Reply:
You could do that. It might be hard to find all of the locations to leave the message at (might not always panic).

Alternatively, I've seen tests that make use of a helper function to check if all required envs are set. If they're not, the test is skipped. This makes it clear that the envs need to be set to successfully run the test.

I've seen something similar to:
```
func testCheckEnvVars(t *testing.T) {
	t.Helper()

	if os.Getenv("AZURE_CLIENT_ID") == "" {
		t.Skip("AZURE_CLIENT_ID not set")
	}
	if os.Getenv("AZURE_CLIENT_SECRET") == "" {
		t.Skip("AZURE_CLIENT_SECRET not set")
	}
	if os.Getenv("AZURE_TENANT_ID") == "" {
		t.Skip("AZURE_TENANT_ID not set")
	}
}
```

Reply:
I'm good with either approach, but I think the experience is better by explicitly telling the person running the test that it's skipped (or even failed) if the envs aren't set.

Reply:
Ya, failing also makes sense. I've seen mixed usage of fail and skip throughout our plugins. Feel free to use fail instead 👍